crates/cargo-bsp/src/cargo_communication/cargo_types/cargo_command.rs

```diff
@@ -6,15 +6,32 @@ use bsp_types::requests::{CompileParams, RunParams, TestParams};
 pub trait CreateCommand {
     fn origin_id(&self) -> Option<String>;
 
-    fn create_command(&self, root: PathBuf) -> Command;
+    fn create_unit_graph_command(&self, root: PathBuf) -> Command;
+
+    fn create_requested_command(&self, root: PathBuf) -> Command;
 }
 
 impl CreateCommand for CompileParams {
     fn origin_id(&self) -> Option<String> {
         self.origin_id.clone()
     }
 
-    fn create_command(&self, root: PathBuf) -> Command {
+    fn create_unit_graph_command(&self, root: PathBuf) -> Command {
+        // TODO add appropriate build target to arguments
+        let mut cmd = Command::new(toolchain::cargo());
+        cmd.current_dir(root)
+            .args([
+                "+nightly",
+                "build",
+                "--unit-graph",
+                "-Z",
+                "unstable-options",
+            ])
+            .args(self.arguments.clone());
+        cmd
```

Comment:
ta logika powtarza się z 6 razy, może warto ją wydzielić do funkcji, na przykład (to tylko pseudokod, nie wiem czy działa):

```
fn create_command(root: PathBuf, custom_args: Vec<&str>, params_args: Vec<String>) -> Command {
    let mut cmd = Command::new(toolchain::cargo());
    cmd.current_dir(root)
        .args(custom_args)
        .args(params_args);
    cmd
}
```

Reply:
Na razie zostawiłam takie komendy, bo Patryk je i tak poprawia

Reply:
okay, to najwyżej w PR Patryka się na to zwróci uwagę